src/lib/fauna-account-client.mjs

```diff
@@ -66,22 +106,26 @@ export class FaunaAccountClient {
       );
       if (response.status >= 400) {
         throw new Error(
-          `Error creating session (${response.status}): ${response.statusText}`
+          `Failure to get session with Fauna (${response.status}): ${response.statusText}`
         );
       }
-      const session = await response.json();
-      return session;
+      return await response.json();
     } catch (err) {
-      throw new Error(
-        "Failure to create session with Fauna: " + JSON.stringify(err)
-      );
+      throw new Error("Failure to get session with Fauna: " + err.message);
     }
   }
 
-  // TODO: remove account_key param and use credential manager helper
-  async listDatabases(account_key) {
+  /**
+   * Lists databases associated with the given account key.
+   *
+   * @param {string} accountKey - The account key to list databases for.
+   * @returns {Promise<Object[]>} - The list of databases.
+   * @throws {Error} - Throws an error if there is an issue during the request.
+   */
+  async listDatabases(accountKey) {
     const headers = new Headers();
-    headers.append("Authorization", `Bearer ${account_key}`);
+    headers.append("Authorization", `Bearer ${accountKey}`);
+
```

Comment:
thoughts on abstracting out something with an API like `makeFaunaRequest` on the next draft of this PR? that'll let you centralize error handling and make these methods really short & similar.